### PR TITLE
Doc: Add Custom Elements

### PR DESCRIPTION
**Pull request by ax3l**
Document how to add custom beamline elements to ImpactX as a user workflow.

- [x] finalize
- [x] rebase after #623 was merged

### CODE REVIEW

File: docs/source/usage/workflows/add_element.rst [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+.. note::
+
+   We plan to add a simple, linear map element that can be configured in user input.
+   Follow `issue #538 <https://github.com/ECP-WarpX/impactx/issues/538>`__ for progress.
+
+
+C++ Element
+-----------
+
```
Comment (cemitch99) [inline docs/source/usage/workflows/add_element.rst:44]: We might want to remark that this workflow is valid for (thin or thick) elements described by explicit maps.  Thick elements with soft-edged fringe fields require additional slicing, to perform integration of the equations for the linear map, before applying this to the particles.  We could say that detailed documentation of implementation of soft-edged elements  is to be addressed later.

**Review by cemitch99 — APPROVED** [review]
This looks great to me.  I made one small comment.

File: docs/source/usage/workflows/add_element.rst (hunk 1 above)
Comment (ax3l) [reply docs/source/usage/workflows/add_element.rst:44]: Good point.

I agree we should elaborate more on how to add thick elements with fringe field in that API. Looking at the current soft-edged elements though: the part we describe here, the push of reference particles and beam particles, is identical, right? I think the detail that differs is what one implements inside the _body_ of the here described APIs, correct?

File: docs/source/usage/workflows/add_element.rst [hunk 2]
```diff
@@ -0,0 +1,87 @@
+.. _usage-workflows-add-element:
+
+Add New Beamline Elements
+=========================
+
+In ImpactX, one can easily add new beamline elements as a user.
+There are multiple ways to add new elements to ImpactX, you can pick the one that fits your needs best.
```
Comment (ax3l) [inline docs/source/usage/workflows/add_element.rst:7]:
```suggestion
There are multiple ways to add new elements to ImpactX, you can pick the one that fits your needs best.

The workflows described here apply both for thin kicks or thick elements.
Thick elements can also use soft-edged fringe fields (see `existing soft-edged elements for implementation details <https://github.com/ECP-WarpX/impactx/tree/development/src/particles/elements>`__).
```

File: docs/source/usage/workflows/add_element.rst (hunk 1 above)
Comment (ax3l) [reply docs/source/usage/workflows/add_element.rst:44]: Added a suggestion above, let me know if that sounds reasonable for now?

File: docs/source/usage/workflows/add_element.rst [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      :language: cpp
+      :dedent: 8
+      :start-at: py::class_<Drift, elements::Thick, elements::Alignment> py_Drift(me, "Drift");
+      :end-at: register_beamoptics_push(py_Drift);
+
+Pull requests that added a new element and can be taken as examples are:
+
+  * `Chromatic Plasma Lens <https://github.com/ECP-WarpX/impactx/pull/514>`__
+  * `Thin-Kick Dipole <https://github.com/ECP-WarpX/impactx/pull/472>`__
+  * `Chromatic Elements for Drift, Quad, Uniform Focusing+Solenoid <https://github.com/ECP-WarpX/impactx/pull/356>`__
```
Comment (ax3l) [inline docs/source/usage/workflows/add_element.rst:89]: Shall we add a soft edge element here to the linked examples?
```suggestion
  * `Chromatic Elements for Drift, Quad, Uniform Focusing+Solenoid <https://github.com/ECP-WarpX/impactx/pull/356>`__
  * `Quadrupole with Soft-Edge Fringe Fields <https://github.com/ECP-WarpX/impactx/pull/322>`__
```

File: docs/source/usage/workflows/add_element.rst (hunk 1 above)
Comment (cemitch99) [reply docs/source/usage/workflows/add_element.rst:44]: Yes, that's true.  Ok, this looks good.

File: docs/source/usage/workflows/add_element.rst (hunk 3 above)
Comment (cemitch99) [reply docs/source/usage/workflows/add_element.rst:89]: Yes, looks good with the soft-edge quadrupole example.

**Review by cemitch99 — APPROVED** [review]
Looks good to me.

**Pull request merged**